app/src/main/java/nl/mpcjanssen/simpletask/task/Priority.kt

```diff
@@ -26,89 +26,38 @@ package nl.mpcjanssen.simpletask.task
+enum class Priority {
 
-    fun inListFormat(): String {
-        return listFormat
-    }
+    A, B, C, D, E, F, G, H, I, J, K, L, M, N, O, P, Q, R, S, T, U, V, W, X, Y, Z,
+    NONE {
+        override val code: String = "-"
+        override val fileFormat: String = ""
+    };
 
-    fun inDetailFormat(): String {
-        return detailFormat
-    }
-
-    fun inFileFormat(): String {
-        return fileFormat
-    }
+    open val code: String = name
+    open val fileFormat: String = "($name)"
 
     companion object {
 
-        private fun reverseValues(): Array<Priority> {
-            val values = Priority.values()
-            return values.reversed().toTypedArray()
-        }
-
-        fun range(p1: Priority, p2: Priority): List<Priority> {
-            val priorities = ArrayList<Priority>()
-            var add = false
-
-            for (p in if (p1.ordinal < p2.ordinal)
-                Priority.values()
-            else
-                Priority.reverseValues()) {
-                if (p == p1) {
-                    add = true
-                }
-                if (add) {
-                    priorities.add(p)
-                }
-                if (p == p2) {
-                    break
-                }
-            }
-            return priorities
-        }
-
-        fun rangeInCode(p1: Priority, p2: Priority): List<String> {
-            val priorities = Priority.range(p1, p2)
-            val result = ArrayList<String>(priorities.size)
-            for (p in priorities) {
-                result.add(p.code)
-            }
-            return result
-        }
+        val codes: List<String>
+            get() = values().map { it.code }.sorted()
 
         fun inCode(priorities: List<Priority>): ArrayList<String> {
-            val strings = ArrayList<String>()
-            for (p in priorities) {
-                strings.add(p.code)
-            }
-            return strings
+            return ArrayList<String>(priorities.map { it.code })
         }
 
-        fun toPriority(codes: List<String>?): ArrayList<Priority> {
-            val priorities = ArrayList<Priority>()
-            if (codes == null) {
-                return ArrayList()
-            }
-            for (code in codes) {
-                priorities.add(Priority.toPriority(code))
-            }
-            return priorities
+        fun toPriority(codes: List<String>): ArrayList<Priority> {
+            return ArrayList<Priority>(codes.map { toPriority(it) })
         }
 
         fun toPriority(s: String?): Priority {
-            if (s == null) {
-                return NONE
+            val upper = s?.toUpperCase(Locale.US) ?: return NONE
+            try {
+                return valueOf(upper)
             }
-
-            for (p in Priority.values()) {
-                if (p.code == s.toUpperCase(Locale.US)) {
-                    return p
-                }
+            catch (e: IllegalArgumentException) {
+                return NONE
```

Comment:
Enums give us a way to convert from strings to instances; use that instead of rolling our own.

The IllegalArgumentException is thrown if there is no enum that corresponds to the string (as you can probably guess).